deps/ccommon/src/cc_itt.c

```diff
@@ -0,0 +1,59 @@
+
+#include <cc_itt.h>
+#include <cc_mm.h>
+#include <cc_log.h>
+#include <cc_debug.h>
+
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+
+#define ITT_MODULE_NAME "ccommon::itt"
+#define ITT_DOMAIN_NAME "cc_itt"
+
+static bool itt_init = false;
+
+#ifdef CC_ITT
+__itt_heap_function cc_itt_malloc;
+__itt_heap_function cc_itt_free;
+__itt_heap_function cc_itt_realloc;
+
+static void
+_cc_initialize(void)
+{
+    cc_itt_malloc = __itt_heap_function_create("allocate", ITT_DOMAIN_NAME);
```

Comment:
I don't like having itt_setup and itt_teardown. Just create these functions on first use. And change the macros into functions for ITT enabled case.